File: src/test/java/com/tibiawiki/domain/repositories/FixtureArticleRepositoryTest.java
package com.tibiawiki.domain.repositories;

import org.junit.jupiter.api.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.notNullValue;
import static org.hamcrest.Matchers.nullValue;

class FixtureArticleRepositoryTest {

    @Test
    void loadsWikiTitlesFromDiskAndNormalizesUnderscores() {
        FixtureArticleRepository repo = new FixtureArticleRepository("regression/fixtures");

        assertThat(repo.getPageNamesFromCategory("Creatures").get(0), is("Dragon"));
        assertThat(repo.getArticle("Dragon"), notNullValue());
        assertThat(repo.getArticle("Carlin_Sword"), notNullValue());
        assertThat(repo.getArticle("ThisDoesNotExistXYZ123"), is(nullValue()));
        assertThat(repo.getArticle("Loot_Statistics:Ferumbras"), notNullValue());
    }
}

File: src/test/kotlin/com/tibiawiki/domain/repositories/FixtureArticleRepositoryTest.kt
package com.tibiawiki.domain.repositories

import com.tibiawiki.domain.objects.WikiNamespace
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.hamcrest.Matchers.notNullValue
import org.hamcrest.Matchers.nullValue
import org.junit.jupiter.api.Test

class FixtureArticleRepositoryTest {

    @Test
    fun loadsWikiTitlesFromDiskAndNormalizesUnderscores() {
        val repo = FixtureArticleRepository("src/test/resources/wiki-fixtures")

        assertThat(repo.getPageNamesFromCategory("Creatures")[0], `is`("Dragon"))
        assertThat(repo.getArticle("Dragon"), notNullValue())
        assertThat(repo.getArticle("Carlin_Sword"), notNullValue())
        assertThat(repo.getArticle("ThisDoesNotExistXYZ123"), `is`(nullValue()))
        assertThat(repo.getArticle("Loot_Statistics:Ferumbras"), notNullValue())
    }

    @Test
    fun namespacedCategoryLookupUsesTheSameFixtureList() {
        val repo = FixtureArticleRepository("src/test/resources/wiki-fixtures")

        assertThat(
            repo.getPageNamesFromCategory("Creatures", WikiNamespace.LOOT_STATISTICS),
            `is`(repo.getPageNamesFromCategory("Creatures"))
        )
    }
}
